pkgs/development/python-modules/cdxj-indexer/default.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  wheel,
+  warcio,
+  surt,
+  idna,
+  py3amf,
+  multipart,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "cdxj-indexer";
+  version = "1.4.6";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "webrecorder";
+    repo = "cdxj-indexer";
+    rev = "v${version}";
+    hash = "sha256-E3b/IfjngyXhWvRYP9CkQGvBFeC8pAm4KxZA9MwOo4s=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
+  ];
+
+  dependencies = [
+    warcio
+    surt
+    idna
+    py3amf
+    multipart
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [
+    "cdxj_indexer"
+  ];
+
+  # Upstream specifies idna<3.0, but the issue that triggered the pinning
+  # has since been fixed: <https://github.com/webrecorder/cdxj-indexer/issues/26>
+  #
+  # Furthermore, we do not enable the signing feature at the moment.
+  dontCheckRuntimeDeps = true;
```

If idna is the (only?) problem then we can use `pythonRelaxDeps` to loosen requiremnts hence builds. or any other reasons?

```suggestion
  pythonRelaxDeps = [
    "idna"
  ];
```

---

I looked a bit deeper, and upstream appears to engage in the questionable practice of pinning transitive dependencies to solve package conflicts, and there are also dependencies that are completely unused. Overall it's a bit disappointing.